homeassistant/components/hausbus/__init__.py

```diff
@@ -0,0 +1,43 @@
+"""The Haus-Bus integration."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .config_entry import HausbusConfig, HausbusConfigEntry
+from .const import DOMAIN
+from .gateway import HausbusGateway
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: HausbusConfigEntry) -> bool:
+    """Set up Haus-Bus from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    gateway = HausbusGateway(hass, entry)
+    #    hass.data[DOMAIN][entry.entry_id] = gateway
+    entry.runtime_data = HausbusConfig(gateway)
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    _LOGGER.debug("start searching devices")
+
+    # search devices after adding all callbacks to the gateway object
+    gateway.home_server.searchDevices()
```

Comment:
If this sends UDP packages, this should be run in an executor:
```suggestion
    hass.async_add_executor_job(gateway.home_server.searchDevices)
```

Also, shouldn't this happen periodically, every minute or something instead of once? What happens if the device is disconnected when Home Assistant starts?

Reply:
Yes, if the Hausbus LAN-Gateway is not connected during Home Assistant startup, all Hausbus devices will be shown as unavailable until you reload the integration. I am unsure if the overhead of a periodic job is worth to handle this corner case, or if it would be enough to add this detail to the documentation. Is there an easy way to schedule a periodic job in home assistant?

Other Hausbus devices on the bus can be detected at runtime, even if they are connected or boot later.

Reply:
It's not OK to require Home Assistant to be restarted.

You write "the Hausbus LAN-Gateway", does it mean a user will always have exactly one gateway, or could a user also have no gateway or more than one gateway?

Reply:
> you can reload the integration from the integrations page

It should not be necessary for the user to perform such manual steps (restart Home Assistant or reload the integration) for a basic and common issue where the device is not connected to the network when Home Assistant starts.

> They need to have one gateway connected to their LAN network. There are multiple devices that can serve as a gateway.

What happens if there are more than one gateway on the LAN network, will Home Assistant connect to one of them based on the UDP discovery?
What happens if the connection to the gateway is lost?

Reply:
Since the communication is UDP there is no long lasting connection. It will just send and receive command and events from the network. If the user connects multiple gateways for some reason, the events would be received multiple times, but this is no problem.
If the LAN gateway got disconnected the commands will not reach the destination and there will be no state update events. So the devices are frozen in the last state from Home Assistant perspective.

Reply:
I also implemented a recurring background job for device discovery. I hope I did it in the correct way.

Reply:
> Since the communication is UDP there is no long lasting connection. It will just send and receive command and events from the network.

Yeah, that's clear 👍 

>  If the user connects multiple gateways for some reason, the events would be received multiple times, but this is no problem.

OK, so if there are more than one gateway on the LAN, the single hausbus config entry take care of them all?

> If the LAN gateway got disconnected the commands will not reach the destination and there will be no state update events. So the devices are frozen in the last state from Home Assistant perspective.

Right, that makes sense. But once they're back online, everything will again work from the user's perspective without manual intervention?

> I also implemented a recurring background job for device discovery. I hope I did it in the correct way.

It looks good 👍

Reply:
> OK, so if there are more than one gateway on the LAN, the single hausbus config entry take care of them all?
Yes.

>  But once they're back online, everything will again work from the user's perspective without manual intervention?
Correct.